system_tests/retry.py

```diff
@@ -4,52 +4,177 @@
 import six
 
 
-class Retry(object):
-    """Retry class for retrying eventually consistent resources in testing."""
+def _retry_all(ignored):
+    """Retry all caught exceptions."""
+    return True
 
-    def __init__(self, exception, tries=4, delay=3, backoff=2, logger=None):
-        """Retry calling the decorated function using an exponential backoff.
 
-        :type exception: Exception or tuple of Exceptions
-        :param exception: The exception to check or may be a tuple of
-                          exceptions to check.
+class RetryBase(object):
+    """Base for retrying calling a decorated function w/ exponential backoff.
 
-        :type tries: int
-        :param tries: Number of times to try (not retry) before giving up.
+    :type max_tries: int
+    :param max_tries: Number of times to try (not retry) before giving up.
 
-        :type delay: int
-        :param delay: Initial delay between retries in seconds.
+    :type delay: int
+    :param delay: Initial delay between retries in seconds.
 
-        :type backoff: int
-        :param backoff: Backoff multiplier e.g. value of 2 will double the
-                        delay each retry.
+    :type backoff: int
+    :param backoff: Backoff multiplier e.g. value of 2 will double the
+                    delay each retry.
 
-        :type logger: logging.Logger instance
-        :param logger: Logger to use. If None, print.
-        """
-
-        self.exception = exception
-        self.tries = tries
+    :type logger: logging.Logger instance
+    :param logger: Logger to use. If None, print.
+    """
+    def __init__(self, max_tries=4, delay=1, backoff=2, logger=None):
+        self.max_tries = max_tries
         self.delay = delay
         self.backoff = backoff
         self.logger = logger.warning if logger else six.print_
 
+
+class RetryErrors(RetryBase):
+    """Decorator for retrying given exceptions in testing.
+
+    :type exception: Exception or tuple of Exceptions
+    :param exception: The exception to check or may be a tuple of
+                        exceptions to check.
+
+    :type error_predicate: function, takes caught exception, returns bool
+    :param error_predicate: Predicate evaluating whether to retry after a
+                            caught exception.
+
+    :type max_tries: int
+    :param max_tries: Number of times to try (not retry) before giving up.
+
+    :type delay: int
+    :param delay: Initial delay between retries in seconds.
+
+    :type backoff: int
+    :param backoff: Backoff multiplier e.g. value of 2 will double the
+                    delay each retry.
+
+    :type logger: logging.Logger instance
+    :param logger: Logger to use. If None, print.
+    """
+    def __init__(self, exception, error_predicate=_retry_all,
+                 max_tries=4, delay=1, backoff=2, logger=None):
```

Comment:
Since this copies the defaults from the base, shouldn't we define those as module-level constants?